instrumentation-api-annotation-support/src/test/groovy/io/opentelemetry/instrumentation/api/annotation/support/MethodCacheTest.groovy

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.annotation.support
+
+import spock.lang.Specification
+
+import java.lang.reflect.Method
+import java.util.function.Function
+
+class MethodCacheTest extends Specification {
```

Comment:
These tests look like they'd be easier to maintain in Java

Reply:
I can port them.  To be honest Spock still confuses the heck out of me, especially mocking interactions.  😅